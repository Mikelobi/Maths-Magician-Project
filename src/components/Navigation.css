.header {
  display: flex;
  flex-direction: row;
  align-items: baseline;
  background-color: rgb(87, 67, 67);
  color: #fff;
  justify-content: space-between;
}

.link {
  text-decoration: none;
  color: #fff;
  padding: 0.5em;
  font-size: 1.3em;
  letter-spacing: 0.05em;
}

.navigation {
  display: flex;
  list-style-type: none;
}
