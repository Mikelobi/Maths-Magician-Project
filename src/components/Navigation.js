import { Link } from 'react-router-dom';
import './Navigation.css';

function Navigation() {
  return (
    <div className="header">
      <h2>Math Magicians</h2>
      <nav className="navigation">
        <Link to="/">Home</Link>
        {' '}
        |
        {' '}
        <Link to="/calculator">Calculator</Link>
        {' '}
        |
        {' '}
        <Link to="/quotes">Quotes</Link>
      </nav>
    </div>
  );
}

export default Navigation;
