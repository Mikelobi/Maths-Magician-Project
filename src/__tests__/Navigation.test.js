import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Navigation from '../components/Navigation';

describe('Test presence of navigation header', () => {
  test('renders the navigation correctly', () => {
    const navComponent = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>,
      )
      .toJSON();
    expect(navComponent).toMatchSnapshot();
  });
});

describe('Test whether links lead to expected path', () => {
  test('Should load Home component', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Home'));
    expect(screen.getByTestId('1'));
  });
});
