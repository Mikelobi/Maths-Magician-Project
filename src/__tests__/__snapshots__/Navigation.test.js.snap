// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of navigation header renders the navigation correctly 1`] = `
<div
  className="header"
>
  <div>
    <h1
      className="header-title"
    >
      Math Magicians
    </h1>
  </div>
  <nav
    className="navigation"
  >
    <li
      className="navLink"
    >
      <a
        aria-current="page"
        className="link active"
        data-testid={1}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navLink"
    >
      <a
        className="link"
        data-testid={2}
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="navLink"
    >
      <a
        className="link"
        data-testid={3}
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </nav>
</div>
`;
